File: medplum-java/medplum-java-model/src/main/java/com/fhir/SubstancePolymer_RepeatUnit.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableSubstancePolymer_RepeatUnit.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface SubstancePolymer_RepeatUnit {
  Optional<List<Extension>> extension();

  Optional<List<SubstancePolymer_StructuralRepresentation>> structuralRepresentation();

  Optional<String> id();

  Optional<List<Extension>> modifierExtension();

  Optional<SubstanceAmount> amount();

  Optional<CodeableConcept> orientationOfPolymerisation();

  Optional<List<SubstancePolymer_DegreeOfPolymerisation>> degreeOfPolymerisation();

  Optional<String> repeatUnit();

  static ImmutableSubstancePolymer_RepeatUnit.Builder builder() {
    return ImmutableSubstancePolymer_RepeatUnit.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/SubstancePolymer_RepeatUnit.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { SubstanceAmount } from "./SubstanceAmount";
import { SubstancePolymer_DegreeOfPolymerisation } from "./SubstancePolymer_DegreeOfPolymerisation";
import { SubstancePolymer_StructuralRepresentation } from "./SubstancePolymer_StructuralRepresentation";

export interface SubstancePolymer_RepeatUnit {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    orientationOfPolymerisation: CodeableConcept | null | undefined;
    repeatUnit: string | null | undefined;
    amount: SubstanceAmount | null | undefined;
    degreeOfPolymerisation:
        | SubstancePolymer_DegreeOfPolymerisation[]
        | null
        | undefined;
    structuralRepresentation:
        | SubstancePolymer_StructuralRepresentation[]
        | null
        | undefined;
}
